lib/grinder_files.dart

```diff
@@ -98,12 +100,210 @@ class FileSet {
+  /**
+   * Creates a temporary directory in the system temp directory. See
+   * [Directory.systemTemp] and [Directory.createTempSync]. If [prefix] is
+   * missing or null, the empty string is used for [prefix].
+   */
+  static Path createSystemTemp([String prefix]) {
+    return new Path(Directory.systemTemp.createTempSync(prefix));
+  }
+
+  static Path get cwd => new Path(Directory.current);
+
+  final String _path;
+
+  /**
+   * Create a new [Path]. The [entity_or_path] parameter can be a
+   * [FileSystemEntity] or a [String].
+   */
+  Path(entity_or_path) : _path = _coerce(entity_or_path);
+
+  String get name {
+    int index = _path.lastIndexOf(_sep);
+    return index != -1 ? _path.substring(index + 1) : null;
+  }
+
+  String get path => _path;
+
+  FileSystemEntity get entity {
+    final FileSystemEntityType type = FileSystemEntity.typeSync(_path);
+
+    if (type == FileSystemEntityType.FILE) {
+      return new File(_path);
+    } else if (type == FileSystemEntityType.DIRECTORY) {
+      return new Directory(_path);
+    } else if (type == FileSystemEntityType.LINK) {
+      return new Link(_path);
+    } else {
+      return null;
+    }
+  }
+
+  bool get exists {
+    return FileSystemEntity.typeSync(_path) != FileSystemEntityType.NOT_FOUND;
+  }
+
+  /**
+   * Returns the containing [Path]. Returns a non-null value even if this is a
+   * root directory.
+   *
+   * See [FileSystemEntity.parent].
+   */
+  Path get parent {
+    int index = _path.lastIndexOf(_sep);
+
+    // Do string manipulation if there are path separators; otherwise, use the
+    // file system entity information.
+    if (index == 0 || index == -1) {
+      FileSystemEntity e = entity;
+      return e == null ? null : new Path(e.parent);
+    } else {
+      return new Path(_path.substring(0, index));
+    }
+  }
+
+//  /**
+//   * Returns the abolute version of this Path.
+//   */
+//  Path get absolute {
+//    // TODO:
+//  }
+
+  bool get isDirectory => FileSystemEntity.isDirectorySync(_path);
+  bool get isFile => FileSystemEntity.isFileSync(_path);
+  bool get isLink => FileSystemEntity.isLinkSync(_path);
+
+  /**
+   * Assume the current file system entity is a [File] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  File get asFile => new File(path);
+
+  /**
+   * Assume the current file system entity is a [Directory] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  Directory get asDirectory => new Directory(path);
+
+  /**
+   * Assume the current file system entity is a [Link] and return it as such.
+   * You would call this instead of [entity] when the file system entity does
+   * not yet exist.
+   */
+  Link get asLink => new Link(path);
+
+  /**
+   * Copy the the entity to the given destination. Return the newly created
+   * [Path].
+   */
+  Path copy(Path destDir) {
```

Comment:
Right now I'm thinking that you'll convert things into Paths, operate on them as Paths, and then convert them out, back into file system entities.